pkgs/by-name/wi/wildcard/package.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, blueprint-compiler
+, cargo
+, desktop-file-utils
+, fetchFromGitLab
+, libadwaita
+, meson
+, ninja
+, pkg-config
+, rustPlatform
+, rustc
+, stdenv
+, wrapGAppsHook4
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wildcard";
+  version = "0.3.3";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Wildcard";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-jOv0l1vnfDePWF7SAbsBFipPAONliPdc47xj79BJ+rc=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-RWEUS7WA/mKCjHLUvkdzsHGg9xQrEVGnvqncBTy1U8c=";
+  };
+
+  nativeBuildInputs = [
```

Comment:
`nativeBuildInputs` means the architecture of the package is the same of the architecture of the build environment, which should be packages that are only available in build time. `buildInputs` are packages (that should be) available at runtime, but they are not simply listed as dependencies, and whether it can become the final runtime dependencies is determined by whether the result has a relationship (linking, including path, etc) with it.

See https://nixos.org/manual/nixpkgs/unstable/#ssec-stdenv-dependencies

Reply:
In this case, libadwaita should go to `buildInputs`, and others are still `nativeBuildInputs`. It's worth noting that enabling `strictDeps = true` can sometimes help you point out these mistakes.

Reply:
Thank you for explanation.